Source/Core/Core/PowerPC/JitILCommon/IR.cpp

```diff
@@ -1199,6 +1203,21 @@ InstLoc IRBuilder::FoldFallBackToInterpreter(InstLoc Op1, InstLoc Op2)
 	return EmitBiOp(FallBackToInterpreter, Op1, Op2);
 }
 
+std::pair<InstLoc, InstLoc> IRBuilder::FoldMemoryAddress(InstLoc addr)
+{
+	if (getOpcode(*addr) == Add && isImm(*getOp2(addr)))
+	{
+		// We have Base + Offset which can be used directly as a memory address.
+		// Forget about the add and use the operands directly
+		return std::make_pair(getOp1(addr), getOp2(addr));
+	}
+	else
+	{
+		// Otherwise we just use the result of the add with a zero offset
+		return std::make_pair(addr, EmitIntConst(0));
```

Comment:
we could implement a constant pooling system so multiple memory address point at the same zero.

But that's probably better done as a generic common expression pass.